clients/src/main/java/org/apache/kafka/common/protocol/types/TaggedFields.java

```diff
@@ -165,7 +165,7 @@ public Map<Integer, Object> validate(Object item) {
             }
             return objects;
         } catch (ClassCastException e) {
-            throw new SchemaException("Not a NavigableMap.");
+            throw new SchemaException("Not a NavigableMap.", e);
```

Comment:
Is this similar to the CompactArrayOf? Generally, if we can provide the right information without having a long chain of exceptions, it's better.

Reply:
That's a good tip Ismael. Thank you! I have fixed this in the latest commit.